docs/designers-developers/developers/internationalization.md

```diff
@@ -1,47 +1,159 @@
+			{ style: { color:'red'}, },
+			__('Hello World', 'myguten')
+		);
+	},
 
-WordPress 5.0 introduces a new function called `wp_set_script_translations( 'my-script-handle', 'my-text-domain' )` to load translation files for a given script handle.
+	save: () => {
+		return el(
+			'p',
+			{ style: { color:'red'}, },
+			__('Hello World', 'myguten')
+		);
+	}
+});
+```
 
-You can learn more about it in [the JavaScript I18N dev note](https://make.wordpress.org/core/2018/11/09/new-javascript-i18n-support-in-wordpress/).
+In the above example, the function will use the first argument for the string to be translated. The second argument is the text domain which must match the text domain slug specified by your plugin.
 
-## More Resources
+Common functions available, these mirror their PHP counterparts are:
+
+- `__( 'Hello World', 'my-text-domain' )` - Translate a certain string.
+- `_n( '%s Comment', '%s Comments', numberOfComments, 'my-text-domain' )` - Translate and retrieve the singular or plural form based on the supplied number.
+- `_x( 'Default', 'block style', 'my-text-domain' )` - Translate a certain string with some additional context.
+
+**Note:** Every string displayed to the user should be wrapped in an i18n function.
+
+After all strings in your code is wrapped, the final step is to tell WordPress your JavaScript contains translations, using the [wp_set_script_translations()](https://developer.wordpress.org/reference/functions/wp_set_script_translations/) function.
+
+```php
+<?php
+	...
+
+	// Added in init action
+	// Uses script handle defined in register
+	wp_set_script_translations( 'myguten-script', 'myguten' );
+```
+
+This is all you need to make your plugin JavaScript code translatable.
+
+When you set script translations for a handle WordPress will automatically figure out if a translations file exists on translate.wordpress.org, and if so ensure that it's loaded into wp.i18n before your script runs.  With translate.wordpress.org, plugin authors also do not need to worry about setting up their own infrastructure for translations and can rely on a global community with dozens of active locales.  Read more about [WordPress Translations](https://make.wordpress.org/meta/handbook/documentation/translations/).
+
+## Provide Your Own Translations
+
+You can create and ship your own translations with your plugin, if you have sufficient knowledge of the language(s) you can ensure the translations are available.
+
+### Create Translation File
+
+The translation files must be in the JED 1.x JSON format.
+
+To create a JED translation file, first you need to extract the strings from the text.
+Using the [wp-cli tool](https://wp-cli.org/), you create a `.pot` file using the following command from within your plugin directory:
+
+```
+wp i18n make-pot ./
+```
+
+This will create the file `myguten.pot` a pot file is a template file to be used for translating to different languages. The pot file is made up of translation sets, for example:
```

Comment:
```suggestion
This will create the file `myguten.pot` which contains all the translatable strings from your project. A single entry in this POT file might look like this:
```